velox/expression/tests/ExprTest.cpp

```diff
@@ -2828,3 +2860,21 @@ TEST_F(ExprTest, lambdaWithRowField) {
 
   assertEqualVectors(array, evalResult);
 }
+
+TEST_F(ExprTest, getVectorFunction) {
+  exec::registerStatefulVectorFunction(
+      "test_vector_func",
+      PlusConstantFunction::signatures(),
+      createTestVectorFunction);
+  auto func = exec::getVectorFunction(
+      "test_vector_func", {}, {}, CppToType<int32_t>::create());
```

Comment:
The signature is expecting an integer argument. So let's provide that. The factory can depend on `inputs[1].type` which is the return type.